pkg/serviceimport/map.go

```diff
@@ -59,8 +58,9 @@ func (si *serviceInfo) resetLoadBalancing() {
 }
 
 type Map struct {
-	svcMap map[string]*serviceInfo
-	mutex  sync.RWMutex
+	svcMap             map[string]*serviceInfo
+	weightFetchingFunc func(service, namesapce, inCluster string) int64
```

Comment:
I suggest to define a type:

`type WeightFunc(service, namespace, cluster string) int64`

```suggestion
	getWeightFor WeightFunc
```